cuda_bindings/tests/test_path_finder.py

```diff
@@ -0,0 +1,85 @@
+import os
+import subprocess  # nosec B404
+import sys
+
+import pytest
+
+from cuda.bindings import path_finder
+from cuda.bindings._path_finder import supported_libs
+
+ALL_LIBNAMES = path_finder._SUPPORTED_LIBNAMES + supported_libs.PARTIALLY_SUPPORTED_LIBNAMES_ALL
+ALL_LIBNAMES_LINUX = path_finder._SUPPORTED_LIBNAMES + supported_libs.PARTIALLY_SUPPORTED_LIBNAMES_LINUX
+ALL_LIBNAMES_WINDOWS = path_finder._SUPPORTED_LIBNAMES + supported_libs.PARTIALLY_SUPPORTED_LIBNAMES_WINDOWS
+if os.environ.get("CUDA_BINDINGS_PATH_FINDER_TEST_ALL_LIBNAMES", False):
+    if sys.platform == "win32":
+        TEST_FIND_OR_LOAD_LIBNAMES = ALL_LIBNAMES_WINDOWS
+    else:
+        TEST_FIND_OR_LOAD_LIBNAMES = ALL_LIBNAMES_LINUX
+else:
+    TEST_FIND_OR_LOAD_LIBNAMES = path_finder._SUPPORTED_LIBNAMES
+
+
+def test_all_libnames_linux_sonames_consistency():
+    assert tuple(sorted(ALL_LIBNAMES_LINUX)) == tuple(sorted(supported_libs.SUPPORTED_LINUX_SONAMES.keys()))
+
+
+def test_all_libnames_windows_dlls_consistency():
+    assert tuple(sorted(ALL_LIBNAMES_WINDOWS)) == tuple(sorted(supported_libs.SUPPORTED_WINDOWS_DLLS.keys()))
+
+
+def test_all_libnames_libnames_requiring_os_add_dll_directory_consistency():
+    assert not (set(supported_libs.LIBNAMES_REQUIRING_OS_ADD_DLL_DIRECTORY) - set(ALL_LIBNAMES_WINDOWS))
+
+
+def test_all_libnames_expected_lib_symbols_consistency():
+    assert tuple(sorted(ALL_LIBNAMES)) == tuple(sorted(supported_libs.EXPECTED_LIB_SYMBOLS.keys()))
+
+
+def _build_subprocess_failed_for_libname_message(libname, result):
+    return (
+        f"Subprocess failed for {libname=!r} with exit code {result.returncode}\n"
+        f"--- stdout-from-subprocess ---\n{result.stdout}<end-of-stdout-from-subprocess>\n"
+        f"--- stderr-from-subprocess ---\n{result.stderr}<end-of-stderr-from-subprocess>\n"
+    )
+
+
+@pytest.mark.parametrize("api", ("find", "load"))
+@pytest.mark.parametrize("libname", TEST_FIND_OR_LOAD_LIBNAMES)
+def test_find_or_load_nvidia_dynamic_library(info_summary_append, api, libname):
+    if api == "find":
+        code = f"""\
+from cuda.bindings._path_finder.find_nvidia_dynamic_library import find_nvidia_dynamic_library
+abs_path = find_nvidia_dynamic_library({libname!r})
+print(f"{{abs_path!r}}")
+"""
```

Comment:
Do these need to be run as subprocesses? Defining these as strings is not ideal for maintenance.

Reply:
I added a comment to explain: commit f3361848437509fd11ba89e5af3e20dcc5b4d590

Yes, subprocesses are necessary here to isolate global dynamic library loading state. Without subprocesses, one test could affect the next (e.g., via already-loaded handles).